charts/intel-gpu-resource-driver/values.yaml

```diff
@@ -0,0 +1,85 @@
+# Default values for intel-gpu-resource-driver.
+nameOverride: ""
+namespaceOverride: ""
+fullnameOverride: ""
+selectorLabelsOverride: {}
+
+crds:
+  enabled: true
```

Comment:
Is this used? AFAIK, CRDs cannot be disabled when they are defined under 'crds' directory in Helm.

Reply:
If crds.enabled is set to` true,` it usually means the CRDs in the crds folder are installed or managed with the Helm chart. If it's `false`, the CRDs are not managed by the Helm chart, useful when they're already installed or managed independently.

Reply:
CRDs are not templated so I think having the value here doesn't apply. Based on the documentation, CRD installation can only be skipped from command line:
https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-1-let-helm-do-it-for-you

Reply:
And according to Helm (v3) documentation, even more worrying is that Helm just will not touch already installed CRDs, regardless of whether they would need to be upgraded or deleted.   There's no option to upgrade obsolete CRDs with Helm.

Best summary I found from the various Helm tickets on the subject, was this: https://github.com/helm/helm/issues/10585#issuecomment-1109561457

Reply:
The `crds.enabled: true` setting in the values.yaml file of a Helm chart is there to make sure all the necessary CRDs are automatically installed for users, especially for those who might not be very familiar with CRDs. If you set it to `false`, it tells the chart not to install or update CRDs, which is useful for people who prefer to handle CRDs themselves. Normally, CRDs are stored in a `crds/ `folder in the chart and are installed by default unless you choose to skip them. This setting gives you more control over that process.

You can see a similar setup in the [kube-prometheus-stack chart](https://github.com/prometheus-community/helm-charts/blob/ee637d09aa6b57e4f0ffcf4d5488c34ca11ff10a/charts/kube-prometheus-stack/values.yaml#L31C1-L34C16).

For our chart, I removed this line and added a note in the README.md that CRDs should be installed separately

Reply:
kube-prometheus-stack chart has a more complex setup. They have a dependency to a subchart that only includes the CRDs: https://github.com/prometheus-community/helm-charts/tree/ee637d09aa6b57e4f0ffcf4d5488c34ca11ff10a/charts/kube-prometheus-stack/charts/crds
And that is enabled or disabled based on the Chart.yaml's conditions based on the values (crds.enabled):
https://github.com/prometheus-community/helm-charts/blob/ee637d09aa6b57e4f0ffcf4d5488c34ca11ff10a/charts/kube-prometheus-stack/Chart.yaml#L44

I think that's a nice way to overcome the restrictions on Helm. In any case, I think it's fine now that the `crds.enabled` is removed.